network/builder/src/builder.rs

```diff
@@ -327,23 +322,24 @@ impl NetworkBuilder {
             None
         };
 
-        // union pubkeys from addrs with pubkeys directly in config
-        let addr_pubkeys_iter = seed_addrs.iter().map(|(peer_id, addrs)| {
-            let pubkey_set: HashSet<_> = addrs
-                .iter()
-                .filter_map(NetworkAddress::find_noise_proto)
-                .collect();
-            (*peer_id, pubkey_set)
-        });
-        for (peer_id, pubkey_set) in addr_pubkeys_iter {
-            seed_pubkeys.entry(peer_id).or_default().extend(pubkey_set);
-        }
+        // Merge pubkeys that may be in the seed addresses
+        // TODO: This seems like a poor choice to hide known keys in the addresses, and possibly even problematic, it's kept for backwards compatibility
+        let mut seeds = seeds.clone();
+        seeds.values_mut().for_each(
```

Comment:
what happens if you do
`let seeds = seeds.clone().values_mut().for_each(`
alternatively why not just use a map function and do the cloning inside?

Reply:
I just dumped this entirely.

Now, whenever a `TrustedPeer` is created, it just searches the addresses for keys, especially since this is what we're using it for every time.